pkg/inventory/storage.go

```diff
@@ -35,6 +37,14 @@ type Storage interface {
 	Store(objs object.ObjMetadataSet, status []actuation.ObjectStatus) error
 	// GetObject returns the object that stores the inventory
 	GetObject() (*unstructured.Unstructured, error)
+	// Apply applies the inventory object. This utility function is used
+	// in InventoryClient.Merge.
+	Apply(dynamic.Interface, meta.RESTMapper, StatusPolicy) error
```

Comment:
Do we need both Apply & Store? Can they both be merged into apply? Or do we explicitly need one that merges and one that replaces?

Reply:
After digging and attempting to refactor both methods into one, there are references to `Store()` outside of the `Apply()` method logic that might make it more difficult to combine both logic. Within kpt as well, there is some usage of `Store()`. I'll opt for keeping it explicitly separate for now for simplicity. LMKYT